docs/blog/2023-03-31-zephyr-3-2.md

```diff
@@ -0,0 +1,303 @@
+author_title: Project Creator
+author_url: https://gitlab.com/petejohanson
+author_image_url: https://www.gravatar.com/avatar/2001ceff7e9dc753cf96fcb2e6f41110
+tags: [firmware, zephyr, core]
+---
+
+I'm happy to announce that we have completed the [work](https://github.com/zmkfirmware/zmk/pull/1499) to upgrade ZMK to [Zephyr 3.2](https://docs.zephyrproject.org/3.2.0/releases/release-notes-3.2.html)!
+
+[petejohanson] did the upgrade work to adjust ZMK for the Zephyr changes, with help from [Nicell] on the LVGL pieces.
+
+- Upgrade to LVGL 8.x API, and move to the new Kconfig settings.
+- Tons of RP2040 work.
+- Zephyr core API changes, including DTS `label` use changes.
+- Move to [pinctrl](https://docs.zephyrproject.org/3.2.0/hardware/pinctrl/index.html) Zephyr subsystem.
+
+## Getting The Changes
+
+Use the following steps to update to the latest tooling in order to properly use the new ZMK changes:
+
+### User Config Repositories Using GitHub Actions
+
+Existing user config repositories using Github Actions to build will pull down Zephyr 3.2 automatically, however if you created your user config a while ago, you may need to update it to reference our shared build configuration to leverage the correct Docker image.
+
+1.  Replace the contents of your `.github/workflows/build.yaml` with:
+
+    ```
+    on: [push, pull_request, workflow_dispatch]
+
+    jobs:
+        build:
+        uses: zmkfirmware/zmk/.github/workflows/build-user-config.yml@main
+    ```
+
+1.  Add a new file to your repository named `build.yaml`:
+
+        ```
+        # This file generates the GitHub Actions matrix
+        # For simple board + shield combinations, add them
+        # to the top level board and shield arrays, for more
+        # control, add individual board + shield combinations to
+        # the `include` property, e.g:
+        #
+        # board: [ "nice_nano_v2" ]
+        # shield: [ "corne_left", "corne_right" ]
+        # include:
+        #   - board: bdn9_rev2
+        #   - board: nice_nano_v2
+        #     shield: reviung41
+        #
+        ---
+        ```
+
+and then update is as appropriate to build the right shields/boards for your configuration.
+
+### Upgrade a manual script
+
+If you have a custom GitHub Actions workflow you need to maintain for some reason, you can update the workflow to to use the `stable` Docker image tag for the build:
+
+- Open `.github/workflows/build.yml` in your editor/IDE
+- Change `zmkfirmware/zmk-build-arm:2.5` to `zmkfirmware/zmk-build-arm:stable` wherever it is found
+- Locate and delete the lines for the DTS output step, which is no longer needed:
+
+  ```
+    - name: ${{ steps.variables.outputs.display-name }} DTS File
+      if: ${{ always() }}
+      run: |
+        if [ -f "build/zephyr/${{ matrix.board }}.pre.tmp" ]; then cat -n build/zephyr/${{ matrix.board }}.pre.tmp; fi
+        if [ -f "build/zephyr/zephyr.dts" ]; then cat -n build/zephyr/zephyr.dts; fi
+  ```
+
+:::note
```

Comment:
This note probably isn't necessary anymore given the previous section instructions, right?

Reply:
Agreed. Removed. I had copied from the previous Zephyr upgrade blog post.